src/core/tsi/ssl_transport_security.c

```diff
@@ -49,6 +49,8 @@
 #include <openssl/x509.h>
 #include <openssl/x509v3.h>
 
+#include "src/core/support/env.h"
```

Comment:
First off, thanks for this patch. I'm aware of the issue and I appreciate you taking a stab at it (I've been a bit overloaded lately...).

However this introduces a dependency on a non-public include. I'd really like to keep TSI (transport security interface) free of any GRPC deps, GPR (Google Portable Runtime) public includes are OK since I think GPR will eventually ship solo but I'm trying hard not to depend on anything else. 

Also, an environment variable is not necessarily a good way to configure the need for a client cert on the server side. You may have 2 different ports in your application, one that needs mutual auth, and the other that does not (or 2 services running on the same process).

I have thought about this and the right way to do this is to either.
- Add an `enforce_mutual_auth` parameter in `grpc_ssl_server_credentials_create` (https://github.com/grpc/grpc/blob/master/include/grpc/grpc_security.h#L171) and plumb it through TSI basically reusing some code that you have here. This would cause an API breakage but I think that it's OK as we have not frozen the APIs yet).
- use the brand new `grpc_call_auth_context` API and fail if the client is not authenticated. @yang-g is working on the C++ wrapping of this interface.

Let me know what you think and thanks again.